wordpress-mobile/WordPress-Android
# Update "Followers" terminology to "Subscribers"

Fixes #20690 
This PR updates `followers` and `follows` terminology in the app to `subscribers`. Text related to following tags or conversations should not be affected, as well as any other usage of `follow` that doesn't fit within the `subscribers` context.

-----

## To Test:
1. To test, check `Stats`, `Notifications`, `Reader`, and `People` since this is where most of the changes occur. 

Extra: Review strings.xml for usages of `followers`, `follows`, `follow`, and `following` to ensure all cases have been updated.

Some examples:

|                                | Before | After |
|--------------------------------|--------|-------|
| Stats -> Insights tab -> No Subscribers |  <img width="382" alt="Screenshot 2024-05-06 at 6 40 37 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/642b3939-56b1-4fa7-b50e-23ef3ffa3dec">    | <img width="224" alt="Screenshot 2024-05-06 at 4 26 34 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/4c8cd5ea-01a7-4711-9ba4-cc0d83dc7fff">   |
| Stats -> Insights tab -> Subscribers    |  <img width="378" alt="Screenshot 2024-05-06 at 6 42 37 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/a14e0928-2bef-455b-8aed-111966a0e8f3">      |    <img width="218" alt="Screenshot 2024-05-06 at 6 02 12 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/973dce1d-0ad0-4dc8-a0b6-c5e099df1ecc"> |
| Notifications -> Follows tab   |   <img width="396" alt="Screenshot 2024-05-06 at 6 44 28 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/cddbc5f2-49d8-4e1d-81ca-7926b74d8bd9">     |    <img width="233" alt="Screenshot 2024-05-06 at 5 53 42 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/bf76ecb9-a108-4b22-a50f-7e2ce9c30e49">  |
| My Site -> More -> People -> `+` button  |      <img width="288" alt="Screenshot 2024-05-06 at 6 31 59 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/d60cb13e-33eb-4a2f-a066-8bcb2f13c758">  |  <img width="221" alt="Screenshot 2024-05-06 at 6 32 09 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/6c6bd10c-445e-405c-ae92-66a42ebb1963">    |
| My Site -> More -> People -> `Team` dropdown menu  |    <img width="387" alt="Screenshot 2024-05-06 at 6 34 45 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/4574815a-91e4-4c36-9e93-ca0929266c67">    |    <img width="319" alt="Screenshot 2024-05-06 at 6 34 35 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/a9e5f944-dbdd-41f0-af60-1b62557abfe0">  |




-----

## Regression Notes

1. Potential unintended areas of impact

    - String translations

3. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

4. What automated tests I added (or what prevented me from doing so)

    - Updated existing tests

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
